src/ledger/mod.rs

```diff
@@ -105,6 +106,54 @@ impl Ledger {
         Ok(())
     }
 
+    #[instrument(name = "ledger.utxo_dropped", skip(self, tx))]
+    pub async fn utxo_dropped(
+        &self,
+        tx: Transaction<'_, Postgres>,
+        tx_id: LedgerTransactionId,
+        detected_txn_id: LedgerTransactionId,
+        ledger_account_ids: WalletLedgerAccountIds,
+    ) -> Result<(), LedgerError> {
+        let txs = self
+            .inner
+            .transactions()
+            .list_by_ids(std::iter::once(detected_txn_id))
+            .await?;
+        if let Some(UtxoDetectedMeta {
```

Comment:
This should not be conditional. If its missing its an error that needs bubbling up.